### PR TITLE
[New Rule] SSH Process Launched From Inside A Container

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/2787
### Details
This rule detects an SSH or SSHD process executed from inside a container. This includes both the client ssh binary and server ssh daemon process. SSH usage inside a container should be avoided and monitored closely when necessary. With valid credentials an attacker may move laterally to other containers or to the underlying host through container breakout. They may also use valid SSH credentials as a persistence mechanism.

### Query
```
process where container.id: "*" and event.type== "start" and
event.action in ("fork", "exec") and event.action != "end" and 
process.name: ("sshd", "ssh", "autossh")
```

<details>
<summary/> Screenshot of Query </summary>

image

</details>

### CODE REVIEW

**Review by terrancedejesus — APPROVED** [review]
LGTM!

- Had a conversation with @imays11 about possibly limiting this PR to `orchestrator.resource.type: pod` as SSH into a node may be commonly observed. Decided to keep the rule as is and let telemetry dictate if that is necessary.

File: rules/integrations/cloud_defend/lateral_movement_ssh_process_launched_inside_a_container.toml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+]
+risk_score = 73
+rule_id = "03a514d9-500e-443e-b6a9-72718c548f6c"
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Lateral Movement", "Persistence", "Container"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where container.id: "*" and event.type== "start" and
```
Comment (Mikaayenson) [inline rules/integrations/cloud_defend/lateral_movement_ssh_process_launched_inside_a_container.toml:39]: Why filter by container.id * if it's all container ids?

**Review by Mikaayenson — APPROVED** [review]
Small nit question. otherwise LGTM

File: rules/integrations/cloud_defend/lateral_movement_ssh_process_launched_inside_a_container.toml (hunk 1 above)
Comment (imays11) [reply rules/integrations/cloud_defend/lateral_movement_ssh_process_launched_inside_a_container.toml:39]: This is the field we're using to keep these rules from triggering on normal linux hosts or VMs, similar to the addition of `host.os` to detection rules

**Pull request merged**